boxsdk/object/folder.py

```diff
@@ -142,6 +143,52 @@ def create_upload_session(self, file_size, file_name):
             response_object=response,
         )
 
+    def chunked_upload_with_stream(self, content_stream, file_size, file_name):
+        """
+        Instantiate the chunked upload instance and create upload session.
+
+        :param content_stream:
+            File-like object containing the content of the part to be uploaded.
+        :type content_stream:
+            :class:`File`
+        :param file_size:
+            The size of the file that this part belongs to.
+        :type file_size:
+            `int`
+        :param file_name:
+            The optional new name of the file
+        :type file_name:
+            `unicode` or None
+        :returns:
+            A :class:`ChunkedUpload` object.
+        :rtype:
+            :class:`ChunkedUpload`
+        """
+        upload_session = self.create_upload_session(file_size, file_name)
+        return ChunkedUpload(upload_session, content_stream, file_size)
+
+    def chunked_upload_with_path(self, file_path):
+        """
+        Instantiate the chunked upload instance and create upload session with path to file.
+
+        :param file_path:
+            The local path to the file you wish to upload.
+        :type file_path:
+            `unicode`
+        :returns:
+            A :class:`ChunkedUpload` object.
+        :rtype:
+            :class:`ChunkedUpload`
+        """
+        total_size = os.stat(file_path).st_size
+        content_stream = open(file_path, 'rb')
+        file_name = os.path.basename(file_path)
+        return self.chunked_upload_with_stream(
+            content_stream=content_stream,
+            file_size=total_size,
+            file_name=file_name
```

Comment:
This will always rename the file based on whatever is on disk — this behavior should 1) be optional, and 2) not be the default.